dtfs-central-api/src/v1/controllers/tfm/deal/tfm-update-deal.controller.js

```diff
@@ -67,9 +76,17 @@ const updateDeal = async (dealId, dealChanges, existingDeal) => {
 
       dealUpdate.tfm.lastUpdated = new Date().valueOf();
     }
+    const updateQuery = {
+      ...$.flatten(withoutId(dealUpdate)),
+      $set: {
+        ...$.flatten(withoutId(dealUpdate)).$set,
+        auditDetails: generateTfmUserAuditDetails(sessionUser?._id), 
```

Comment:
Probably want to delete optional chaining eventually, but currently lots of tests fail otherwise